From 70085c8d42e7cf046bfcba276bea9ee1495812c2 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Mon, 10 Jan 2022 09:24:19 +0200
Subject: [PATCH] [SW-69650] gaudi2: pci: improve pll workaround

Function hl_pll_clk_switch_to_pll() can handle the switching sequence
from any clock, so there is no need to call hl_pll_clk_switch_to_ref()
before it.

Change-Id: I12af62de59c8babb3c8a48106e1f656f2aff69bd
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/gaudi2/common/asic.c b/arch/gaudi2/common/asic.c
index 4d68720..aa1a49b 100644
--- a/arch/gaudi2/common/asic.c
+++ b/arch/gaudi2/common/asic.c
@@ -228,7 +228,6 @@
 	 * After phy fw upgrade is complete and pci link is up we move div0
 	 * to his nominal clock frequency.
 	 */
-	hl_pll_clk_switch_to_ref(PCI_PLL_BASE, HL_PLL_DIV0);
 	hl_pll_clk_switch_to_pll(PCI_PLL_BASE, HL_PLL_DIV0, HL_PLL_PLL_CLK);
 }
 
